### ISSUE TITLE
Resolver Templates

### ISSUE THREAD

**Issue by polus-arcticus** [CoopHive/alkahest-mocks#1]
resolver templates for making certain obligations/comments, as off-chain or delegated attestations

I see this as a fork a github template (like a uniswap v4) https://github.com/uniswapfoundation/v4-template that is foundry (later hardhat if wanted, etc) and is basically like this current repo state but with caps locked fill in with your own info ( or later perhaps a cli project generator) that can do some filling in automatically.

**Comment by mlegls** [comment]
Yeah, I like this example. The protocol specification (including implementation guidelines) will have to go beyond just a set of abstract contracts, since Solidity's type system isn't powerful enough to make invalid implementations unrepresentable. Examples would definitely help towards this, and maybe a test suite too, and maybe eventually automatic linters

**Comment by mlegls** [comment]
moved to hooks-based architecture for obligation & escrow contracts
